program/rust/build.rs

```diff
@@ -7,7 +7,33 @@ use {
 };
 
 fn main() {
-    println!("cargo:rustc-link-search=./program/c/target");
+    // Cargo exposes ENV variables for feature flags that can be used to determine whether to do a
+    // self-contained build that compiles the C components automatically.
+    if std::env::var("CARGO_FEATURE_LIBRARY").is_ok() {
```

Comment:
is there any reason not to do this for both build paths? this seems like a great simplification of the current build process.

Reply:
No reason, this is the correct way to do it. But I wasn't sure how many things would break and this was a safe small PR to get a review on. I can make it the default and check anywhere that might break / has been relying on relative paths. I just didn't plan to do it in this PR.

Reply:
Let's get this merged and I can make it the default